### ISSUE TITLE
Add smoke test before uploading release

### ISSUE THREAD

**Issue by codefromthecrypt** [tetratelabs/archive-envoy#31]
Right now, we don't test the operating system matrix works before uploading a release. We should check what we expect to work, noting that this could be impacted by what envoy supports. For example, we have to be careful that cutting a new version like 1.21, which expands the OS version matrix, that doesn't prevent cutting a patch of an older version like 1.19. Also, not all operating systems may be available at the same time. For this reason, a manual target could be acceptable instead, that runs the checks.

TL;DR; a manually triggered workflow that runs all operating systems we want to expect, building as necessary that just invokes `envoy version` will be helpful. For example `envoy version` segfaults currently on macOS 12 and that's easier to start with vs a more elaborate test (see #28)

**Comment by codefromthecrypt** [comment]
Note: this won't be able to test macOS 12 because no major CI supports it, yet, and as far as I know, there's no way to run it in docker.

GHA: https://github.com/actions/runner/issues/1171
CircleCI: https://circleci.com/docs/2.0/testing-ios/#supported-xcode-versions
Travis: https://docs.travis-ci.com/user/reference/osx/

**Comment by codefromthecrypt** [comment]
https://github.com/tetratelabs/archive-envoy/pull/32 tests what we can, but doesn't block releases as there are reasons why things might not pass (ex homebrew PR not merged). The important thing is we will know what worked or not.
